bins/bins-extra.sh

```diff
@@ -144,6 +144,62 @@ download_file() {
     checksum ${output} $2 || false
 }
 
+download_docker_image() {
+    # set extra filename output or default output
+    if [ ! -z $2 ]; then
+        output=$2
+    else
+        output=$(basename "$1").tar
+    fi
+
+    # set default url
+    fileurl=$1
+
+    docker pull "${fileurl}"
+    docker save "${fileurl}" -o "${output}"
+}
+
+extract_docker_image() {
+    # you need to run `docker save <image> -o <file>.tar
+    # then run this script like `extrat-image.sh <file>.tar`
+    # it will create a directoyr `<file>.tar-root` which containers
```

Comment:
directoyr -> directory, which -> where

Reply:
I will re-write the entire description because when i started i had something else in mind